addons/project/models/project_task.py

```diff
@@ -1104,11 +1105,12 @@ def _compute_partner_id(self):
             if not task.partner_id:
                 task.partner_id = self._get_default_partner_id(task.project_id, task.parent_id)
 
-    @api.depends('project_id')
+    @api.depends('project_id', 'parent_id.milestone_id')
     def _compute_milestone_id(self):
         for task in self:
-            if (task.project_id or task.parent_id.project_id) != task.milestone_id.project_id:
-                task.milestone_id = False
+            if task.milestone_id and task.project_id == task.milestone_id.project_id:
+                continue
+            task.milestone_id = task.parent_id.project_id == task.project_id and task.parent_id.milestone_id
```

Comment:
```suggestion
            if task.milestone_id and task.project_id == task.milestone_id.project_id:
                continue
            task.milestone_id = task.parent_id.milestone_id if task.parent_id.project_id == task.project_id and task.parent_id.milestone_id
```
Shouldn't it be like that ? otherwise it would only get a boolean value

Reply:
Ugh you are right, I had completely forgotten that you can do that :sweat_smile: Eh, I never really liked this assignment syntax because it confused me but we have used it before and its concise so good job - keep it as is :grin: